std-bits/table/src/main/java/org/enso/table/data/column/builder/LongBuilderChecked.java

```diff
@@ -28,14 +25,34 @@ protected LongBuilderChecked(
         new CastProblemAggregator(problemAggregator, relatedColumnName, type);
   }
 
+  @Override
+  public void appendLong(long value) {
+    if (!type.fits(value)) {
+      castProblemAggregator.reportNumberOutOfRange(value);
+      appendNulls(1);
+      return;
+    }
+    super.appendLong(value);
+  }
+
   @Override
   public void appendNoGrow(Object o) {
+    if (o != null) {
+      Long x = NumericConverter.tryConvertingToLong(o);
+      if (x != null) {
+        if (type.fits(x)) {
+          appendLong(x);
+          return;
+        }
+        castProblemAggregator.reportNumberOutOfRange(x);
+      }
+    }
     if (o == null) {
-      isNothing.set(currentSize++);
+      appendNulls(1);
     } else {
```

Comment:
Something is off here. The `else` branch here executes if `o != null`. That is completely redundant with the first `if` at the top of `appendNoGrow`. It almost seems like the method body is duplicated. But the one below is wrong as it does not do bounds checking.

Reply:
Yes looks like I mucked that one up when was adjusting in last commit.